### PR TITLE
upgrade go to 1.20

### PR DESCRIPTION
**Pull request by siddarthkay**
## Summary

This PR also points to status-go branch where we have upgraded go to 1.20
Related statusgo PR -> https://github.com/status-im/status-go/pull/5027

### Testing notes
Please test everything, specially the store node stuff.

#### Platforms
- Android
- iOS

status: ready

### CODE REVIEW

File: status-go-version.json
Comment (shivekkhurana) [inline status-go-version.json]: Shouldn't this be a tagged release on go side too?

Comment (siddarthkay) [reply status-go-version.json]: it should be, atm its just to see if builds pass or fail.
I'm awaiting approvals on status-go side.
I'll mark this PR as draft for now :)

Comment (siddarthkay) [reply status-go-version.json]: got the approvals, I'll push a tag on status-go side once QA approves this PR.
Thats the process -> https://github.com/status-im/status-mobile/blob/develop/doc/merging-pr-process.md#status-go-changes

**Comment by VolodLytvynenko** [comment]
Hi @siddarthkay thank you for PR. Take a look at a found issue

## PR_ISSUE 1: Community is not fetched

### Steps:
1. `User A` Creates a community (current community is created within current PR build)
2. `User A` shares community with `User B`
3. `User B` opens community link

### Actual result:
Community is not fetched

https://github.com/status-im/status-mobile/assets/52490791/b6089c63-9f8d-44d9-b0e9-58938eff66bb

### Expected result:
Community is fetched

### OS: 
IOS, Android

### Logs
[logs (10).zip](https://github.com/status-im/status-mobile/files/14944111/logs.10.zip)

**Comment by siddarthkay** [comment]
Thanks for testing @VolodLytvynenko : we're tracking the progress of Issue 1 here -> https://github.com/status-im/status-mobile/issues/19581

**Comment by siddarthkay** [comment]
Rebase from top of Eiffel Tower

**Review by shivekkhurana — APPROVED** [review]

**Comment by siddarthkay** [comment]
Hi @VolodLytvynenko : Some additional info regarding ISSUE 1, this happens only on Android.
On iOS I am able to fetch community.

Screenshot 2024-04-12 at 7 05 51 PM

**Comment by siddarthkay** [comment]
needs more work to figure out storenode stuff

**Comment by siddarthkay** [comment]
@status-im/mobile-qa : ISSUE 1 should be fixed, ready for retesting
Tested locally, works for me™️
Screenshot_20240417-171917

**Comment by VolodLytvynenko** [comment]
hi @siddarthkay thank you for the fixes. PR is ready to be merged

File: status-go-version.json
Comment (siddarthkay) [reply status-go-version.json]: tag pushed, status-go PR is merged.

**Review by jakubgs — COMMENTED** [review]
Good work figuring out the fix.

**Pull request merged**
